internal/impl/update.go

```diff
@@ -20,19 +21,37 @@ func (d *Devbox) Update(ctx context.Context, pkgs ...string) error {
 		pkgsToUpdate = d.Packages()
 	}
 
-	for _, pkg := range pkgsToUpdate {
-		if !lock.IsVersionedPackage(pkg) {
+	inputs := nix.InputsFromStrings(pkgsToUpdate, d.lockfile)
+	for _, pkg := range inputs {
+		if pkg.IsLegacy() {
+			fmt.Fprintf(d.writer, "Updating %s -> %s\n", pkg.Raw, pkg.LegacyToVersioned())
+			if err := d.Remove(ctx, pkg.Raw); err != nil {
+				return err
+			}
+			if err := d.lockfile.ResolveToCurrentNixpkgCommitHash(
+				pkg.LegacyToVersioned(),
+			); err != nil {
+				return err
+			}
+			if err := d.Add(ctx, pkg.LegacyToVersioned()); err != nil {
```

Comment:
`ResolveToCurrentNixpkgCommitHash` will resolve the new "LegacyToVersioned" version of the package (i.e. `curl` -> `curl@latest`) to the same hash that was previously being used (either the default or one specified in config).

When `Add` is called, it will resolve to whatever is in lockfile (`Add` does not update version if package is already pinned in lockfile). This behavior mimics `yarn`. So there is no overwriting.

Reply:
ah, I see. Thanks 🙏 

So, the purpose of Add here to ensure we add the newly resolved package to the nix profile?

Reply:
nix profile, devbox.json and any other things we go when we add (create wrappers, plugins, etc)